plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codemodernizer/model/CodeModernizerArtifact.kt

```diff
@@ -99,7 +100,17 @@ open class CodeModernizerArtifact(
             if (!patchesDir.isDirectory()) {
                 throw RuntimeException("Expected root for patches was not a directory.")
             }
-            return patchesDir.walk()
+            val patchPathSequence = patchesDir.walk()
+            val patchFileList = patchPathSequence.map { fileSystem.findFileByNioFile(it) }.toList()
+            if (patchFileList[0] == null) {
+                ApplicationManager.getApplication().invokeAndWait {
+                    ApplicationManager.getApplication().runWriteAction {
+                        // try refreshing Virtual File System if diff.patch not found
+                        VirtualFileManager.getInstance().refreshAndFindFileByNioPath(patchPathSequence.first())
+                    }
+                }
+            }
+            return patchPathSequence
                 .map { fileSystem.findFileByNioFile(it) ?: throw RuntimeException("Could not find patch") }
```

Comment:
Leos advice: replace line 114 `map` with `refreshAndFindFileByNioPath` instead of `findFileByNioFile`

Reply:
Leos comment: We also only return a single patch file, if the `walk()` complicates things, just check array index [0] directly